File: tests/adapters/rust/goldens/imports/strip_external.rs
// Rust module for testing import optimization.

// Standard library imports (external)
// … 23 imports omitted

// More standard library
// … 5 imports omitted

// Third-party library imports (external)
// … 16 imports omitted

// Database crates
// … 4 imports omitted

// Web framework crates
// … 7 imports omitted

// Local/project imports (should be considered local)
use crate::models::User;
use crate::services::UserService;
use crate::database::Connection;
use crate::errors::{ValidationError, NetworkError};
use crate::utils::helpers::{date_formatter, json_parser};
use crate::types::{ApiResponse, UserModel, PostModel};

// Relative imports with different depth levels
use super::utilities;
use super::super::core::core_module;
use super::super::super::config::app_config;

// Long import lists from single module (candidates for summarization)
use crate::validation::{
    email_validator,
    password_validator,
    phone_validator,
    postal_code_validator,
    credit_card_validator,
    input_sanitizer,
    currency_formatter,
    phone_formatter,
    slug_generator,
    hash_creator,
    hash_verifier,
};

use crate::operations::{
    create_user,
    update_user,
    delete_user,
    get_user_by_id,
    get_user_by_email,
    get_users_by_role,
    get_users_with_pagination,
    activate_user,
    deactivate_user,
    reset_user_password,
    change_user_role,
    validate_user_permissions,
};

struct ImportTestService {
    user_service: UserService,
    db_connection: Connection,
    logger: Box<dyn std::any::Any>,
}

impl ImportTestService {
    fn new(user_service: UserService, db_connection: Connection, logger: Box<dyn std::any::Any>) -> Self {
        Self {
            user_service,
            db_connection,
            logger,
        }
    }

    fn process_data(&self, data: Vec<Box<dyn std::any::Any>>) -> Vec<HashMap<String, String>> {
        // Using standard library
        let mut processed = Vec::new();

        for item in data {
            let mut result = HashMap::new();
            // Using chrono for timestamps
            let timestamp = Utc::now().to_rfc3339();
            result.insert("timestamp".to_string(), timestamp);
            processed.push(result);
        }

        // Using local utilities (would call validation functions here)
        processed
    }

    async fn make_http_request(&self, url: &str) -> Result<String> {
        // Using reqwest
        let client = Client::builder()
            .timeout(Duration::from_secs(5))
            .user_agent("ImportTestService/1.0")
            .build()?;

        let response = client.get(url).send().await?;

        if !response.status().is_success() {
            anyhow::bail!("HTTP request failed with status: {}", response.status());
        }

        let text = response.text().await?;
        Ok(text)
    }

    fn serialize_data(&self, data: &dyn std::any::Any) -> Result<String> {
        // Using serde_json
        let json_value = json!({
            "data": "test"
        });

        Ok(serde_json::to_string_pretty(&json_value)?)
    }

    async fn query_database(&self, sql: &str) -> Result<Vec<Row>> {
        // Using sqlx
        let pool = PgPool::connect("postgresql://localhost/test").await?;

        let rows = sqlx::query(sql).fetch_all(&pool).await?;

        Ok(rows)
    }

    async fn start_server(&self) -> Result<()> {
        // Using actix-web
        HttpServer::new(|| {
            App::new()
                .route("/health", web::get().to(|| async { HttpResponse::Ok().body("OK") }))
        })
        .bind(("127.0.0.1", 8080))?
        .run()
        .await?;

        Ok(())
    }
}

// Forward declarations (should not be treated as imports)
struct Forward;
mod module_forward;

File: tests/adapters/java/goldens/imports/strip_external.java
/**
 * Java module for testing import optimization.
 */

package com.example.imports;

// … 37 imports omitted (41 lines)

// Local/relative imports (should be considered local)
import com.example.imports.services.UserService;
import com.example.imports.database.DatabaseConnection;
import com.example.imports.errors.ValidationError;
import com.example.imports.errors.NetworkError;
import com.example.imports.utils.helpers.DateFormatter;
import com.example.imports.utils.helpers.JsonParser;
import com.example.imports.types.ApiResponse;
import com.example.imports.types.UserModel;
import com.example.imports.types.PostModel;

// Imports from different package levels
import com.example.shared.SharedUtility;
import com.example.core.CoreModule;
import com.example.config.AppConfig;

// … 3 imports omitted (4 lines)
import static com.example.imports.utils.Constants.*;

// … 12 imports omitted (13 lines)

import com.example.imports.services.createUser;
import com.example.imports.services.updateUser;
import com.example.imports.services.deleteUser;
import com.example.imports.services.getUserById;
import com.example.imports.services.getUserByEmail;
import com.example.imports.services.getUsersByRole;
import com.example.imports.services.getUsersWithPagination;
import com.example.imports.services.activateUser;
import com.example.imports.services.deactivateUser;
import com.example.imports.services.resetUserPassword;
import com.example.imports.services.changeUserRole;
import com.example.imports.services.validateUserPermissions;

import com.example.utils.validation.EmailValidator;
import com.example.utils.validation.PasswordValidator;
import com.example.utils.validation.PhoneNumberValidator;
import com.example.utils.validation.PostalCodeValidator;
import com.example.utils.validation.CreditCardValidator;
import com.example.utils.validation.InputSanitizer;
import com.example.utils.validation.CurrencyFormatter;
import com.example.utils.validation.PhoneNumberFormatter;
import com.example.utils.validation.SlugGenerator;
import com.example.utils.validation.HashCreator;
import com.example.utils.validation.HashVerifier;

@Service
public class ImportTestService {
    private static final Logger logger = LoggerFactory.getLogger(ImportTestService.class);

    @Autowired
    private UserService userService;

    @Autowired
    private DatabaseConnection dbConnection;

    @Autowired
    private ObjectMapper objectMapper;

    public ApiResponse<List<Map<String, Object>>> processData(List<Object> data) throws JsonProcessingException {
        // Using external libraries
        List<Map<String, Object>> processed = data.stream()
            .map(item -> {
                Map<String, Object> result = new HashMap<>();
                result.put("id", UUID.randomUUID().toString());
                result.put("timestamp", Instant.now().toString());
                result.put("item", item);
                return result;
            })
            .collect(toList());

        // Using local utilities
        List<Map<String, Object>> validated = processed.stream()
            .filter(item -> {
                String email = (String) item.get("email");
                return email != null && EmailValidator.validate(email);
            })
            .collect(toList());

        // Using Java NIO
        Path filePath = Paths.get("output.json");
        String json = objectMapper.writeValueAsString(validated);
        Files.writeString(filePath, json);

        return new ApiResponse<>(
            true,
            validated,
            DateFormatter.format(LocalDateTime.now())
        );
    }

    public String makeHttpRequest(String url) throws IOException, InterruptedException {
        try {
            // Using Java 11+ HTTP Client
            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .timeout(Duration.ofSeconds(5))
                .header("User-Agent", "ImportTestService/1.0")
                .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            return response.body();
        } catch (Exception e) {
            logger.error("HTTP request failed", e);
            throw new NetworkError("Request failed");
        }
    }

    public String serializeData(Object data) throws JsonProcessingException {
        // Using Jackson
        objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
        return objectMapper.writeValueAsString(data);
    }
}

// Annotated classes using validation imports
@Entity
@Table(name = "users")
public class ValidatedUser {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotNull
    private Long id;

    @NotBlank
    @Size(min = 2, max = 100)
    @Column(nullable = false)
    private String name;

    @Email
    @NotEmpty
    @Column(nullable = false, unique = true)
    private String email;

    @Positive
    private Integer age;

    @Pattern(regexp = "^\\+?[1-9]\\d{1,14}$")
    private String phone;

    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public Integer getAge() { return age; }
    public void setAge(Integer age) { this.age = age; }
    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }
}

// REST Controller using Spring annotations
@RestController
@RequestMapping("/api/users")
public class UserController {
    @Autowired
    private ImportTestService importTestService;

    @GetMapping("/{id}")
    public ResponseEntity<ValidatedUser> getUser(@PathVariable Long id) {
        // Implementation
        return ResponseEntity.ok().build();
    }

    @PostMapping
    public ResponseEntity<ValidatedUser> createUser(@Valid @RequestBody ValidatedUser user) {
        // Implementation
        return ResponseEntity.status(CREATED).build();
    }

    @PutMapping("/{id}")
    public ResponseEntity<ValidatedUser> updateUser(
        @PathVariable Long id,
        @Valid @RequestBody ValidatedUser user
    ) {
        // Implementation
        return ResponseEntity.ok().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteUser(@PathVariable Long id) {
        // Implementation
        return ResponseEntity.noContent().build();
    }
}
